mod/item.php

```diff
@@ -992,16 +1005,35 @@ function item_post(&$a) {
 		// Insert an item entry for UID=0 for global entries
 		// We have to remove or change some data before that,
 		// so that the post appear like a regular received post.
-		unset($datarray['self']);
-		unset($datarray['wall']);
-		unset($datarray['origin']);
-
-		if (in_array($datarray['type'], array("net-comment", "wall-comment")))
-			$datarray['type'] = 'remote-comment';
-		elseif ($datarray['type'] == 'wall')
-			$datarray['type'] = 'remote';
-
-		add_shadow_entry($datarray);
+		// Additionally there is some data that isn't a database field.
+		$arr = $datarray;
```

Comment:
I find it more clean since we never know if someone isn't adding some programming later that would rely on the original values. But: I guess I will rebuild this part anyway in an upcoming pull request since I want to work with transactions there.